### PR TITLE
Fix #4419: Remove TODO Marked Files And Other Alpha Kenya Related Files

### LINKED ISSUE #4419: Remove Kenya user study specific constructs once study is over
See TODO locations in code for this issue for places that need cleanup.

### PR DESCRIPTION
**Pull request by XichengSpencer**
## Explanation
Fix #4419 
Removed application\alphakenya package as Adhiambo suggests.
Removed codes in bazel files,test_file_exemptions.textproto that is related with alphakenya 
Remove AlphaKenya related modules under android/util/logging including KenyaAlphaPlatformParameterModule, EventLoggingModule etc. and corresponding tests and Bazel configs
Inlined StandardEventTypeToHumanReadableNameConverterImpl.kt into EventTypeToHumanReadableNameConverter.kt
Removed EventLoggingConfigurationModule

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-A11y-Guide))
- For PRs introducing new UI elements or color changes, both light and dark mode screenshots must be included
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

### CODE REVIEW

**Comment by adhiamboperes** [comment]
@theMr17, could you please take an initial pass at this PR?

**Comment by adhiamboperes** [comment]
@XichengSpencer, there are a number of merge conflicts in this PR, could you please try to resolve them?

**Comment by theMr17** [comment]
Hi @XichengSpencer, assign me the PR for review once the conflicts are resolved.

File: .github/workflows/build_tests.yml
Comment (adhiamboperes) [inline .github/workflows/build_tests.yml]: Yes, should be removed, because this workflow will start failing now that the kenya build has been removed.

File: app/src/main/java/org/oppia/android/app/application/alphakenya/AlphaKenyaOppiaApplication.kt
Comment (adhiamboperes) [inline app/src/main/java/org/oppia/android/app/application/alphakenya/AlphaKenyaOppiaApplication.kt:1]: You need to also delete the BUILD and the AlphaKenyaApplicationComponent, and remove the package entirely.

**Review by adhiamboperes — CHANGES_REQUESTED** [review]
Thanks @XichengSpencer! PTAL at the comment about files that still need to be removed, and also sync your branch with develop.

**Review by adhiamboperes — APPROVED** [review]
Thanks @XichengSpencer! This cleanup looks really good.

It seems you have some conflicts with develop, could you please resolve those by pulling develop into your local branch, then pushing after resolving any conflicts?

**Comment by adhiamboperes** [comment]
It is also good practice to respond to all review comments that you have addressed, then assign the reviwer through the "assignees" section once done.

**Comment by XichengSpencer** [comment]
> It is also good practice to respond to all review comments that you have addressed, then assign the reviwer through the "assignees" section once done.

Thanks for letting me know Adhiambo!

**Comment by XichengSpencer** [comment]
> Thanks @XichengSpencer! This cleanup looks really good.
> 
> It seems you have some conflicts with develop, could you please resolve those by pulling develop into your local branch, then pushing after resolving any conflicts?

Done!

**Comment by XichengSpencer** [comment]
> It is also good practice to respond to all review comments that you have addressed, then assign the reviwer through the "assignees" section once done.

Sorry, I accidentally requested reviewer on the reviewer section, but I also did it on the assignees section.

**Comment by adhiamboperes** [comment]
Thanks @XichengSpencer!

The PR LGTM. However, these changes require code owner approval, and both of the oppia/owners are out temporarily so we won't be able to merge this yet. If during merge there are conflicts to resolve or any other issue to address, @BenHenning will let you know.

Please feel free to work on other issues in the meantime.

Additionally, the "reviewer section" is automatically assigned, so you only ever need to change the "assignees section".

File: utility/src/main/java/org/oppia/android/util/logging/BUILD.bazel
Comment (BenHenning) [inline utility/src/main/java/org/oppia/android/util/logging/BUILD.bazel:156]: Can you also clean up this bit? We no longer need to have a separate interface/implementation for the event type to readable name converter. Specifically:

https://github.com/oppia/oppia-android/blob/develop/utility/src/main/java/org/oppia/android/util/logging/StandardEventTypeToHumanReadableNameConverterImpl.kt

Can be inlined into:

https://github.com/oppia/oppia-android/blob/develop/utility/src/main/java/org/oppia/android/util/logging/EventTypeToHumanReadableNameConverter.kt

I think this will allow removing EventLoggingConfigurationModule as well.

File: version.bzl
Comment (BenHenning) [inline version.bzl]: Let's go ahead and bump these versions right away so that there isn't a gap (that means setting ``OPPIA_GA_VERSION_CODE`` to 167 and then adding one for each version above it).

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks @XichengSpencer! It's great to see this get cleaned up. I had two additional suggestions (one of which is a consequential clean-up that we can do because of these changes). PTAL.

**Comment by adhiamboperes** [comment]
@XichengSpencer, friendly ping to address the reviewer comments on this PR.

File: version.bzl
Comment (XichengSpencer) [reply version.bzl]: Just double confirm you mean OPPIA_GA_VERSION_CODE to 161 or 167? Here are the current codes I have from latest develop branch
"OPPIA_DEV_VERSION_CODE = 166
OPPIA_DEV_KITKAT_VERSION_CODE = 165
OPPIA_ALPHA_VERSION_CODE = 163
OPPIA_ALPHA_KITKAT_VERSION_CODE = 162
OPPIA_BETA_VERSION_CODE = 161
OPPIA_GA_VERSION_CODE = 160"

File: utility/src/main/java/org/oppia/android/util/logging/BUILD.bazel
Comment (XichengSpencer) [reply utility/src/main/java/org/oppia/android/util/logging/BUILD.bazel:156]: I searched the usage for EventLoggingConfigurationModule which has 177 usages in tests and components. I just want to get an confirmation and I can remove them all. Hope removing them all won't cause any problem.

File: version.bzl
Comment (XichengSpencer) [reply version.bzl]: Version code updated

File: utility/src/main/java/org/oppia/android/util/logging/BUILD.bazel
Comment (XichengSpencer) [reply utility/src/main/java/org/oppia/android/util/logging/BUILD.bazel:156]: Additional kenyaAlpha clean done, Inlined to [EventTypeToHumanReadableNameConverter.kt], and clean the EventLoggingConfigurationModule.

**Review by adhiamboperes — CHANGES_REQUESTED** [review]
Hi @XichengSpencer, could you please take a look at the following:
1. Static check failures - kdoc validity and link checks 
2. There seem to be some leftover modules in  some test classes, and the removed module appears to still be available in  a BUILD file.

Please make sure youb reassign the PR once done.

**Comment by XichengSpencer** [comment]
> Hi @XichengSpencer, could you please take a look at the following:
> 
> 1. Static check failures - kdoc validity and link checks
> 2. There seem to be some leftover modules in  some test classes, and the removed module appears to still be available in  a BUILD file.
> 
> Please make sure youb reassign the PR once done.

Clean up done!

File: utility/src/main/java/org/oppia/android/util/logging/BUILD.bazel
Comment (BenHenning) [reply utility/src/main/java/org/oppia/android/util/logging/BUILD.bazel:156]: Ah sorry--missed the earlier comment re: confirmation. I can confirm the approach looks correct!

File: utility/src/main/java/org/oppia/android/util/logging/EventTypeToHumanReadableNameConverter.kt
Comment (BenHenning) [inline utility/src/main/java/org/oppia/android/util/logging/EventTypeToHumanReadableNameConverter.kt]:
```suggestion
 * Converter for event types into loggable, human-readable names.
```

Looks like this is out-of-date now that there's no longer an interface + multiple implementations. The summary fragment from the original interface seems suitable for this, now.

File: utility/src/test/java/org/oppia/android/util/logging/BUILD.bazel
Comment (BenHenning) [inline utility/src/test/java/org/oppia/android/util/logging/BUILD.bazel]: Nit: Please add back EOF newline.

File: scripts/assets/test_file_exemptions.textproto
Comment (BenHenning) [inline scripts/assets/test_file_exemptions.textproto]: Can this exemption now be removed? You technically added a test for it by renaming the standard impl test.

File: version.bzl
Comment (BenHenning) [reply version.bzl]: The latest looks correct--thanks!

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks so much @XichengSpencer! This looks fantastic! Really sorry about the long review delay here.

I just had a few comments. Aside from those, can you please also update to the latest develop and fix the failing CI checks?

File: scripts/assets/test_file_exemptions.textproto
Comment (XichengSpencer) [reply scripts/assets/test_file_exemptions.textproto]: Weirdly, I could find this on my local. Because when I remove a file, I will do a global search to make sure all other appearance is removed. And it has happened more than once in this PR, just confused.
image

Comment (adhiamboperes) [reply scripts/assets/test_file_exemptions.textproto]: @XichengSpencer, I think the exemption @BenHenning highlighted is for `EventTypeToHumanReadableNameConverter`

Comment (XichengSpencer) [reply scripts/assets/test_file_exemptions.textproto]: > @XichengSpencer, I think the exemption @BenHenning highlighted is for `EventTypeToHumanReadableNameConverter`

Got it, thanks Adhiambo!

**Review by adhiamboperes — APPROVED** [review]
Thanks @XichengSpencer, I have looked at your latest changes and they seem to have addressed @BenHenning's comments. Handing over to @BenHenning to resplve.

**Review by BenHenning — APPROVED** [review]
Thanks @XichengSpencer! It's great to see this done. PR LGTM. :)

**Comment by BenHenning** [comment]
Updating to latest develop and enabling auto-merge.

**Pull request merged**
